File: Leetcode/62-Unique-Paths/UniquePaths.js
/**
 * @param {number} m
 * @param {number} n
 * @return {number}
 */
var uniquePaths = function(m, n) {
    if (m<=0 || n<=0) {
        return 0
    }
    var N=n+m-2
    var k=m-1
    var res = 1.0
    for (var i=1; i<=k; i++) {
        res = res * (N - k + i) / i
    }
    return ~~res
};

File: Leetcode/62-Unique-Paths/UniquePaths.go
func uniquePaths(m int, n int) int {
    if n<=0 || m<=0 {
        return 0
    }
    N := float64(n+m-2)
    k := float64(m-1)
    res := float64(1.0)
    for i:=1.0; i<=k; i++ {
        res = res * (N - k + i) / i
    }
    return int(res)
}
